beacon-chain/state/v1/proofs_test.go

```diff
@@ -16,19 +16,41 @@ func TestBeaconStateMerkleProofs(t *testing.T) {
 	htr, err := st.HashTreeRoot(ctx)
 	require.NoError(t, err)
 	t.Run("current sync committee", func(t *testing.T) {
-		_, err := st.CurrentSyncCommitteeProof()
+		_, err := st.CurrentSyncCommitteeProof(ctx)
 		require.ErrorContains(t, "unsupported", err)
 	})
 	t.Run("next sync committee", func(t *testing.T) {
-		_, err := st.NextSyncCommitteeProof()
+		_, err := st.NextSyncCommitteeProof(ctx)
 		require.ErrorContains(t, "unsupported", err)
 	})
 	t.Run("finalized root", func(t *testing.T) {
 		finalizedRoot := st.FinalizedCheckpoint().Root
-		proof, err := st.FinalizedRootProof()
+		proof, err := st.FinalizedRootProof(ctx)
 		require.NoError(t, err)
 		gIndex := v1.FinalizedRootGeneralizedIndex()
 		valid := trie.VerifyMerkleProof(htr[:], finalizedRoot, gIndex, proof)
 		require.Equal(t, true, valid)
 	})
+	t.Run("recomputes root on dirty fields", func(t *testing.T) {
```

Comment:
This doesn't actually test this correctly, as the `HashTreeRoot()` recomputes the root again if there are any dirty fields, so this specific case isnt tested correctly. What might be a better thing to do would be to verify that the proof is valid for the particular state root. Since that is what we ultimately care about. This applies the same to all the different state types.